lib/vfscore/lookup.c

```diff
@@ -96,13 +96,14 @@ namei_follow_link(struct dentry *dp, char *node, char *name, char *fp, size_t mo
 	return (0);
 }
 /*
- * Convert a pathname into a pointer to a dentry
+ * Resolve a pathname into a pointer to a dentry and a realpath.
  *
  * @path: full path name.
  * @dpp:  dentry to be returned.
+ * @realpath: if not NULL, return path after resolving all symlinks
  */
 int
-namei(const char *path, struct dentry **dpp)
+namei_resolve(const char *path, struct dentry **dpp, char realpath[PATH_MAX])
```

Comment:
```suggestion
namei_resolve(const char *path, struct dentry **dpp, char *realpath)
```

Let's have this without the specified size, just like it happens for the other argument.